### PR TITLE
Update dkg script to use `click` options - exec args were getting complex

### PR DESCRIPTION
**Pull request by derekpierre**
**Type of PR:**
- [ ] Bugfix
- [ ] Feature
- [ ] Documentation
- [X] Other

**Required reviews:** 
- [ ] 1
- [x] 2
- [ ] 3

**What this does:**
Update `nucypher_dkg` script to use `click` instead of exec args.

Successfully created rituals # 14 and # 15 on `lynx` using the script.

```
$ python ./scripts/hooks/nucypher_dkg.py --eth-provider <> --coordinator-provider <> --ritual-id -1 --signer keystore://<>
...
--------- Initiating Ritual ---------
Using account 0xfDd99d58a7D5696c4f757fAFcAc401d5c9AB87A3 to initiate DKG Ritual
Enter your keystore password:
Commencing DKG Ritual on mumbai using 0xfDd99d58a7D5696c4f757fAFcAc401d5c9AB87A3
Using staking providers for DKG: ['0x210eeAC07542F815ebB6FD6689637D8cA2689392', '0xb15d5A4e2be34f4bE154A1b08a94Ab920FfD8A41']
Broadcasting INITIATERITUAL Legacy Transaction (0.00022690050453801 ETH @ 1.50000003 gwei)
TXHASH 0xeb8485c8d4bc2d4fb004aed4a88ee4e0eac5b27752f6020143c763a37f44c6e1
Waiting 600 seconds for receipt
DKG Ritual #15 initiated: 0xeb8485c8d4bc2d4fb004aed4a88ee4e0eac5b27752f6020143c763a37f44c6e1
Waiting for Ritual to complete; 0s elapsed thus far
Waiting for Ritual to complete; 20s elapsed thus far
Waiting for Ritual to complete; 40s elapsed thus far
Waiting for Ritual to complete; 60s elapsed thus far
Waiting for Ritual to complete; 80s elapsed thus far
DKG Ritual #15 completed after 101s
--------- Data Encryption ---------
...
```

**Issues fixed/closed:**
> - Fixes #...

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

### CODE REVIEW

File: scripts/hooks/nucypher_dkg.py [hunk 1]
```diff
@@ -36,191 +24,224 @@
...
+    type=click.Choice(["mumbai"]),
+    default="mumbai",
+)
+@click.option(
+    "--ritual-id",
+    "ritual_id",
+    "-r",
+    help="Ritual ID; existing id or -1 (to initiate a new ritual)",
+    type=click.INT,
+    required=True,
```
Comment (manumonti) [inline scripts/hooks/nucypher_dkg.py:63]:
```suggestion
    default=-1,
```

This is just a suggestion. Could it make sense to have a default value of -1?

From UX perspective, it sounds weird to me introducing -1 value when actually I want a new ritual ID to be generated.

With this suggestion, user can use `--ritual-id` argument when there is an initiated ritual, or don't use this argument if there is not an initiated ritual (and therefore, a new ritual will be initiated).

WDYT?

File: scripts/hooks/nucypher_dkg.py [hunk 2]
```diff
@@ -1,21 +1,9 @@
...
-# Go through entire process of initiating a ritual, waiting for it to finish, then use for encryption/decryption:
-#    python ./scripts/hooks/nucypher_dkg.py <ETH_PROVIDER_URI> <COORDINATOR_PROVIDER_URI> -1 <SIGNER_URI_FOR_COORDINATOR_NETWORK>
-#
-
 import random
 import sys
 import time
 
 import click
+import maya
```
Comment (manumonti) [inline scripts/hooks/nucypher_dkg.py:6]: Just out of curiosity: What are the advantages of using maya instead of canonical time here?

**Review by manumonti — APPROVED** [review]
I really like how the code looks with CLICK, the code is pretty clean. Thanks.

LGTM 💥! I've just left some suggestions/questions.

**Review by piotr-roslaniec — APPROVED** [review]

File: scripts/hooks/nucypher_dkg.py (hunk 1 above)
Comment (cygnusv) [reply scripts/hooks/nucypher_dkg.py:63]: Agree with Manu here!

**Review by cygnusv — APPROVED** [review]
Looks good! I'm wondering if we should turn some of our scripts into ape scripts, there may be benefits, like account management.

File: scripts/hooks/nucypher_dkg.py (hunk 1 above)
Comment (derekpierre) [reply scripts/hooks/nucypher_dkg.py:64]: Good point! ✔️

File: scripts/hooks/nucypher_dkg.py (hunk 2 above)
Comment (derekpierre) [reply scripts/hooks/nucypher_dkg.py:6]: `maya` makes displaying the elapsed time in seconds easier - https://github.com/derekpierre/nucypher/blob/clickerize-script/scripts/hooks/nucypher_dkg.py#L176.

`time.time() - start_time` produces a float that needs rounding (eg. `18.45042395591736`) whereas `maya.now() - start_time)` produces a `datetime.timedelta`.

**Comment by derekpierre** [comment]
> Looks good! I'm wondering if we should turn some of our scripts into ape scripts, there may be benefits, like account management.

@cygnusv interested to hear more about what you are thinking - don't know if I can picture it just yet.

File: scripts/hooks/nucypher_dkg.py (hunk 2 above)
Comment (manumonti) [reply scripts/hooks/nucypher_dkg.py:6]: Thanks!

**Pull request merged**